remoteappmanager/static/js/home/controller.js

```diff
@@ -1,136 +1,47 @@
-    dialogs, 
-    analytics, 
-    models, 
-    application_list_view,
-    application_view, 
-    resources,
-    hb,
-    init) {
+    application_list_component,
+    application_view_component,
+    helpers) {
     "use strict";
+    var Status = application_list_component.Status;
 
-    var ga = analytics.init();
-    init.handlebars();
+    // Ember Application
 
-    // This model keeps the retrieved content from the REST query locally.
-    // It is only synchronized at initial load.
-    var model = new models.ApplicationListModel();
-    var app_list_view = new application_list_view.ApplicationListView(model);
-    var app_view = new application_view.ApplicationView(model);
+    var AppList = Ember.Application.create({
+        rootElement: '#ember_container'
+    });
 
-    app_list_view.entry_clicked = function (index) {
-        model.selected_index = index;
-        app_list_view.update_selected();
-        app_view.render(false, 200);
-    };
-    
-    app_list_view.stop_button_clicked = function(index) {
-        if (model.status[index] === models.Status.STOPPING) {
-            return;
-        }
-        model.status[index] = models.Status.STOPPING;
-        
-        var app_info = model.app_data[index];
-        var url_id = app_info.container.url_id;
+    // Add Ember helpers to the application
 
-        app_list_view.update_entry(index);
-        
-        resources.Container.delete(url_id)
-            .done(function () {
-                model.update_idx(index)
-                    .done(function() {
-                        app_list_view.update_entry(index);
-                        app_view.render(true);
-                    })
-                    .fail(function(error) {
-                        model.status[index] = models.Status.STOPPED;
-                        app_list_view.update_entry(index);
-                        app_view.render(true);
-                        dialogs.webapi_error_dialog(error);
-                    });
-                })
-            .fail(
-                function (error) {
-                    model.status[index] = models.Status.STOPPED;
-                    app_list_view.update_entry(index);
-                    app_view.render(true);
-                    dialogs.webapi_error_dialog(error);
-                });
-    };
+    Object.keys(helpers).forEach(function(key) {
+        AppList[key] = helpers[key];
+    });
 
-    app_view.start_button_clicked = function (index) {
-        // The container is not running. This is a start button.
-        var mapping_id = model.app_data[index].mapping_id;
-        var image_name = model.app_data[index].image.name;
+    // Ember Controller
 
-        if (model.status[index] === models.Status.STARTING) {
-            return;
-        }
-        
-        model.status[index] = models.Status.STARTING;
-        app_view.render(false, null);
-        app_list_view.update_entry(index);
+    AppList.ApplicationController = Ember.Controller.extend({
```

Comment:
The main application controller which is the link between the application_list_component and the application_view_component. It only contains the current clicked application in the application list and keep it updated with the application view.